File: Backend appbrc/brandcomputer/src/main/java/com/app/brc/brandcomputer/components/memory_ram/model/MemoryRam.java
package com.app.brc.brandcomputer.components.memory_ram.model;

import com.app.brc.brandcomputer.accounting.nir.model.Nir;
import com.app.brc.brandcomputer.computers.model.Computer;
import com.app.brc.brandcomputer.computersAssembly.model.ComputerAssembly;
import lombok.*;
import org.hibernate.annotations.CreationTimestamp;
import org.hibernate.annotations.UpdateTimestamp;

import javax.persistence.*;
import java.time.LocalDateTime;

@Getter
@Setter
@Builder
@NoArgsConstructor
@AllArgsConstructor
@EqualsAndHashCode
@Entity
@Table(name = "memoryRam")
public class MemoryRam {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private int id;

    @ManyToOne
    @JoinColumn(name = "productCode_id")
    private GenerateProductCodeMemoryRam generateProductCodeMemoryRam;

    private String serialNumber;
    private String manufacturer;
    private String module;
    private String ramType;
    private String frequency;
    private String capacity;

    private Double priceIn;
    private String productInformation;
    private String state;
    private String category;
    private int quantity;
    private String unitOfMeasurement;

    private LocalDateTime soldAt;
    private String soldBy;
    private boolean sold;
    private boolean received;

    @CreationTimestamp
    private LocalDateTime createdDate;
    private String createdBy;

    @UpdateTimestamp
    private LocalDateTime lastUpdated;
    private String updatedBy;

    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "computer_id")
    private ComputerAssembly computerAssembly;
}

File: Frontend appbrc/brandcomputers/src/app/model/MemoryRam.ts
import {GenerateProductCode} from './GenerateProductCode';

export class MemoryRam {
  id: number;
  generateProductCode: GenerateProductCode;
  serialNumber: string;
  manufacturer: string;
  module: string;
  ramType: string;
  frequency: string;
  capacity: string;
  priceIn: number;
  productInformation: string;
  state: string;
  soldAt: Date;
  soldBy: string;
  sold: boolean;
  createdDate: Date;
  createdBy: string;
  lastUpdated: Date;
  updatedBy: string;

  static get generateProductURL(): string {
    return 'generate-product-code-memory-ram';
  }

  static get generateProductURLSearch(): string {
    return 'generate-product-code-memory-ram/search';
  }
}